linera-service/tests/end_to_end_tests.rs

```diff
@@ -1298,26 +1312,144 @@ async fn test_end_to_end_fungible() {
-        .get_fungible_account_owner_amount(&account_owner1)
+    app1.assert_fungible_app_has_accounts([
+        (account_owner1, Amount::from(4)),
+        (account_owner2, Amount::from(0)),
+    ])
+    .await;
+    app2.assert_fungible_app_has_accounts([
+        (account_owner1, Amount::from(0)),
+        (account_owner2, Amount::from(3)),
+    ])
+    .await;
+
+    node_service1.assert_is_running();
+    node_service2.assert_is_running();
+}
+
+#[test_log::test(tokio::test)]
+async fn test_end_to_end_crowd_funding() {
+    use crowd_funding::InitializationArguments;
+    use fungible::{AccountOwner, InitialState};
+    let _guard = INTEGRATION_TEST_GUARD.lock().await;
+
+    let network = Network::Grpc;
+    let mut runner = TestRunner::new(network, 4);
+    let client1 = runner.make_client(network);
+    let client2 = runner.make_client(network);
+
+    runner.generate_initial_validator_config().await;
+    client1.create_genesis_config().await;
+    client2.init(&[]).await;
+
+    // Create initial server and client config.
+    runner.run_local_net().await;
+    let (contract_fungible, service_fungible) = runner.build_application("fungible").await;
+
+    let chain1 = client1.get_wallet().default_chain().unwrap();
+    let client2key = client2.keygen().await.unwrap();
+
+    // Create chain2 using client1.
+    let (effect_id, chain2) = client1.open_chain(chain1, Some(client2key)).await.unwrap();
+
+    // Assign chain2 to client2key.
+    assert_eq!(chain2, client2.assign(client2key, effect_id).await.unwrap());
+
+    // The players
+    let owner1 = client1.get_owner().unwrap();
+    let account_owner1 = AccountOwner::User(owner1);
+    let owner2 = client2.get_owner().unwrap();
+    let account_owner2 = AccountOwner::User(owner2);
+    // The initial accounts on chain1
+    let accounts = BTreeMap::from([(account_owner1, Amount::from(6))]);
+    let state_fungible = InitialState { accounts };
+
+    // Setting up the application fungible
+    let application_id_fungible = client1
+        .publish_and_create(
+            contract_fungible,
+            service_fungible,
+            state_fungible.to_string(),
+            None,
+            vec![],
+            None,
+        )
         .await;
-    assert_eq!(value, Amount::from(4));
 
-    let value = app1
-        .get_fungible_account_owner_amount(&account_owner2)
+    // Setting up the application crowd funding
+    let deadline = Timestamp::from(std::u64::MAX);
+    let target = Amount::from(1);
+    let state_crowd = InitializationArguments {
+        owner: account_owner1,
+        deadline,
+        target,
+    };
+    let (contract_crowd, service_crowd) = runner.build_application("crowd-funding").await;
+    let application_id_crowd = client1
+        .publish_and_create(
+            contract_crowd,
+            service_crowd,
+            state_crowd.to_string(),
+            Some(application_id_fungible.clone()),
+            vec![application_id_fungible.clone()],
```

Comment:
It is really not great to have the `application_id_fungible` showing up in two entries of the `publish_and_create`.

Reply:
I had actually already created a issue for this: #723